### ISSUE TITLE
Memory leak receiving binary datachannel messages

### ISSUE THREAD

**Issue by rosskukulinski** [node-webrtc/node-webrtc#205]
So I've been doing some digging into a memory leak in our application that uses node-webrtc.

We've narrowed the leak down to only when we're receiving binary datachannel messages on node 0.10.36.  When we use websockets instead of datachannels, there is no leak.

I'm not super familiar with using NAN, but I'm wondering if we're not properly cleaning up the incoming message?  https://github.com/js-platform/node-webrtc/blob/develop/src/datachannel.cc#L184

**Comment by modeswitch** [comment]
Thanks for narrowing this down. I'll have a look at the code.

**Comment by mafrost** [comment]
Any news on this? I'm also seeing a possible leak that we've traced to wrtc and this might be it.

**Comment by SourceBoy** [comment]
I also have reasons to believe there may be memory leaks after sending/receiving data via `RTCDataChannel.send()`.

I'm using the latest 0.0.61 & linux-64 wrtc.node binary in Node 6.9.4.

Whenever I transfer a file via `send()`, the resident memory size difference between prior to transfer & after transfer is always about the size of the file, reported both by `process.memoryUsage()` and the OS itself. Closing the data channel and/or the peer connection don't seem to free the memory either, but exiting Node does.

At first I thought it could be Node's stream, but I wrote a barebone test that pipes a `fs.createReadStream()` to a `net.Socket` that sends the file to a `net.Server` with the same file, and the memory usage was fine.
